src/autorun.ts

```diff
@@ -116,57 +123,89 @@ async function setupQUnit (qunitBasePath: string) {
   const qunitJSPath = `${qunitBasePath}qunit.js`
   await import(qunitJSPath)
 
+  ;(QUnit as any).on('error', (error: any) => {
+    testResultErrors.push({
+      message: error?.message,
+      stack: error?.stack
+    })
+  })
   ;(QUnit as any).on('testEnd', (qunitTestEndResult: WTRQUnitTestEndResult) => {
-    const errors = collectErrors(qunitTestEndResult)
-    testResultErrors.push(...errors)
+    addToTestSuiteResults(qunitTestEndResult)
   })
   ;(QUnit as any).on('runEnd', (qunitSuiteResults: WTRQUnitSuiteResult) => {
-    const testResults = collectTestSuiteResults(qunitSuiteResults)
-    sessionFinished({
-      passed: testResultErrors.length === 0,
-      errors: testResultErrors,
-      testResults
-    }).catch((err) => console.error(err))
+    // using a setTimeout here to try and help capture any out of sync assertions prior to calling sessionFinished
```

Comment:
I don't really understand how a delay of 1 millisecond helps here. Do you have an example what case the timeout is trying to fix?

Reply:
It was an attempt to capture [this type of failure](https://github.com/qunitjs/qunit/blob/13f0926119781c55eb4c4512b6aa58ecd7f7b7dd/src/test.js#L553). However, I think it can probably be dropped because we can't really know what a reasonable amount of time to wait is. I think not capturing these types of failures will just be a limitation of using @web/test-runner. Something to include in the readme.

```
QUnit.test('testing', assert => {
  setTimeout(() => {
    assert.true(false, 'test')
  }, 1)
})
```

Reply:
i think you can safely ignore this case because here the test is broken. The correct test case would look something like:

    QUnit.test('testing', assert => {
        const done = assert.async();
        setTimeout(() => {
            assert.true(false, 'test');
            done();
        }, 1)
    })